src/Form/ParagraphsPasteForm.php

```diff
@@ -131,13 +131,23 @@ public static function pasteSubmit(array $form, FormStateInterface $form_state)
       )
     );
 
-    // Split on urls and double newlines.
-    $data = preg_split('~(https?://[^\s/$.?#].[^\s]*|[\r\n]+\s?[\r\n]+)~', $pasted_data, -1, PREG_SPLIT_NO_EMPTY | PREG_SPLIT_DELIM_CAPTURE);
+    $settings = $form_object->getFormDisplay($form_state)->getComponent($submit['field_name'])['third_party_settings']['paragraphs_paste'];
 
-    $property_path_mapping = $form_object->getFormDisplay($form_state)
-      ->getComponent($submit['field_name'])['third_party_settings']['paragraphs_paste']['mapping'];
+    $reg_ex = [];
+    if (!empty($settings['split_method']['paragraph'])) {
+      $reg_ex[] = "[\r\n]+\s?[\r\n]+";
+    }
+    if (!empty($settings['split_method']['oEmbed'])) {
+      $reg_ex[] = "https?://[^\s/$.?#].[^\s]*";
+    }
+    if (!empty($settings['split_method']['regex'])) {
+      $reg_ex[] = $settings['split_method']['regex'];
+    }
+
+    // Split by RegEx.
+    $data = preg_split('~(' . implode('|', $reg_ex) . ')~', $pasted_data, -1, PREG_SPLIT_NO_EMPTY | PREG_SPLIT_DELIM_CAPTURE);
 
-    $items = self::traverseData($data, $property_path_mapping);
+    $items = self::traverseData($data, $settings['mapping']);
```

Comment:
I think the plugin needs information about the split method. 

Say, the stuff was split on "h2:" this is lost on preg_split, mylife plugin would not be able to put the line into  heading field.

Reply:
Shouldn't that be handled in the PR where we try to implement that?

Reply:
Shouldn't that be handled in the PR where we try to implement that?

Reply:
I just thought you might want to keep the naming rather than changing it back in a few PRs ;-)